spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java

```diff
@@ -32,25 +35,48 @@
 @ConfigurationProperties("spring.cloud.gcp.config")
 public class GcpConfigProperties implements CredentialsSupplier {
 
-	/** Enables Spring Cloud GCP Config. */
+	/**
+	 *  Enables Spring Cloud GCP Config.
+	 */
 	private boolean enabled;
 
-	/** Name of the application. */
+	/**
+	 *  Name of the application.
+	 */
+	@Value("${spring.application.name:application}")
 	private String name;
 
-	/** Profile under which app is running (e.g., "prod", "dev", "test", etc.). */
-	private String profile = "default";
+	/**
+	 *  Comma-delimited string of profiles under which the app is running.
```

Comment:
It appears that Spring Cloud Config, parses the list of profiles and applies them all at the same time. Since we do not have that ability on GCP currently, I would suggest instead to just use the last profile in the list, not the whole comma-delimited list.